pkg/virtctl/vm/vm_test.go

```diff
@@ -273,35 +273,81 @@ var _ = Describe("VirtualMachine", func() {
 			})
 		})
 
-		It("should force restart vm", func() {
+		DescribeTable("with flags: ", func(force bool, gracePeriodSet bool) {
+			var cmd *cobra.Command
 			vm := kubecli.NewMinimalVM(vmName)
 
-			kubecli.MockKubevirtClientInstance.EXPECT().VirtualMachine(k8smetav1.NamespaceDefault).Return(vmInterface).Times(1)
-			gracePeriod := int64(0)
-			restartOptions := v1.RestartOptions{
-				GracePeriodSeconds: &gracePeriod,
-				DryRun:             nil,
-			}
-			vmInterface.EXPECT().ForceRestart(context.Background(), vm.Name, &restartOptions).Return(nil).Times(1)
+			if gracePeriodSet != force {
```

Comment:
There is an attempt here to use a common function when it is clearly not a fit.
You need two functions here, one that treats `gracePeriodSet != force` scenarios and one for the other.

A test table is a fit when multiple scenarios/inputs share the same test logic, but here it is clearly not the case.

Reply:
Updated and split up the tests.